File: Data Structures & Algorithms/partition-equal-subset-sum/submission-6.cpp
class Solution {
public:
    bool canPartition(vector<int>& nums) {
        int sum = 0;
        for(int num : nums) sum += num;
        if(sum % 2 == 1) return false;
        vector<int> dp(sum, 0);
        dp[0] = 1;

        for(int num : nums) {
            for(int i = sum/2 ; i >= num ; i--) {
                if(dp[i - num]) {
                    dp[i] = 1;
                }
            }
        }

        return dp[sum / 2];
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # mảng đếm -> max heap k 
        cnt = {}
        for num in nums:
            cnt[num] = cnt.get(num, 0) + 1
        return heapq.nlargest(k, cnt, key=lambda x: cnt[x])
